applications/vanilla/controllers/class.postcontroller.php

```diff
@@ -162,6 +162,10 @@ public function discussion($CategoryUrlCode = '') {
             $this->title(t('New Discussion'));
         }
 
+        if (isset($this->Draft) && $this->Draft->Type) {
```

Comment:
Good suggestion. `$this->setData('Type', val('Type', $this->Draft, null));` could go in the else closure of the `if (isset($this->Discussion))` block.